app-preview/src/main/java/com/mapbox/navigation/examples/preview/dropinui/AddViewAnnotationsActivity.kt

```diff
@@ -0,0 +1,113 @@
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.dropin.map.MapViewObserver
+import com.mapbox.navigation.examples.preview.R
+import com.mapbox.navigation.examples.preview.databinding.MapboxActivityViewAnnotationsBinding
+import com.mapbox.navigation.examples.preview.databinding.MapboxItemViewAnnotationBinding
+import com.mapbox.navigation.ui.utils.internal.ifNonNull
+
+/**
+ * The example demonstrates how to add view annotations on top of map view using `NavigationView`
+ * at runtime.
+ *
+ * Before running the example make sure you have put your access_token in the correct place
+ * inside [app-preview/src/main/res/values/mapbox_access_token.xml]. If not present then add
+ * this file at the location mentioned above and add the following content to it
+ *
+ * <?xml version="1.0" encoding="utf-8"?>
+ * <resources xmlns:tools="http://schemas.android.com/tools">
+ *     <string name="mapbox_access_token"><PUT_YOUR_ACCESS_TOKEN_HERE></string>
+ * </resources>
+ *
+ * The example uses replay location engine to facilitate navigation without physically moving.
+ *
+ * How to use the example:
+ * - Start the example
+ * - Grant the location permissions if not already granted
+ * - You should see a view annotation showing your current location
+ */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class AddViewAnnotationsActivity : AppCompatActivity() {
+
+    private val viewAnnotationMap = mutableMapOf<Point, View>()
+    private var mapView: MapView? = null
+    private lateinit var binding: MapboxActivityViewAnnotationsBinding
+    private val locationObserver = object : LocationObserver {
+        override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
+            ifNonNull(mapView) { mapView ->
+                val location = locationMatcherResult.enhancedLocation
+                createViewAnnotation(
+                    mapView,
+                    Point.fromLngLat(location.longitude, location.latitude)
+                )
+            }
+        }
+
+        override fun onNewRawLocation(rawLocation: Location) {
+            // no impl
+        }
+    }
+    private val mapViewObserver = object : MapViewObserver() {
+        override fun onAttached(mapView: MapView) {
+            super.onAttached(mapView)
+            this@AddViewAnnotationsActivity.mapView = mapView
+        }
+
+        override fun onDetached(mapView: MapView) {
+            super.onDetached(mapView)
+            this@AddViewAnnotationsActivity.mapView = null
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = MapboxActivityViewAnnotationsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.navigationView.registerMapObserver(mapViewObserver)
+        MapboxNavigationApp.current()?.registerLocationObserver(locationObserver)
```

Comment:
These should be registering `MapboxNavigationObserver`, and then register inside an onAttached

```
private val navigationObserver = forwardMapboxNavigation(this::onAttached, this::onDetached)

override fun onCreate(savedInstanceState: Bundle?) {
  MapboxNavigationApp.registerObserver(navigationObserver)

override fun onDestroy(savedInstanceState: Bundle?) {
  MapboxNavigationApp.unregisterObserver(navigationObserver)

private val onAttached(mapboxNavigation: MapboxNavigation) {
  mapboxNavigation.registerLocationObserver(locationObserver)
}

private val onDetached(mapboxNavigation: MapboxNavigation) {
  mapboxNavigation.unregisterLocationObserver(locationObserver)
}
```

Reply:
Done.